nanomi-optics/nanomi_optics/gui/window_main.py

```diff
@@ -175,3 +192,47 @@ def u_lens_mode(self):
         elif md == "Cf":
             self.mode = True
         self.upper_menu.set_mode(self.mode)
+
+    def save_results(self):
+        """save results in csv file"""
+        file_path = tk.filedialog.askdirectory()
+        file_path = file_path + "/nanomi_results.csv"
+        save_csv(
+            self.diagram.cf_u, self.diagram.active_lu,
+            self.diagram.cf_l, self.diagram.active_ll,
+            self.mag_u, self.mag_l, CA_DIAMETER,
+            self.diagram.last_mag, file_path
+        )
+
+    def reset_settings(self):
+        """Reset the software to initial state"""
+        temp_func_update_r = self.update_results
+        temp_func_update_l = self.diagram.update_l_lenses
+        temp_func_update_u = self.diagram.update_u_lenses
+        self.update_results = self.pass_function
+        self.diagram.update_l_lenses = self.pass_function
+        self.diagram.update_u_lenses = self.pass_function
+
+        self.upper_menu.mode_widget.option_var.set("Cf")
+        for toggle in self.upper_menu.toggles:
+            if not toggle.get_status():
+                toggle.click()
+
+        for i, link in enumerate(self.upper_menu.links):
+            link.set(self.upper_menu.focal_values[i])
+
+        self.lower_menu.lens_sel.set(-1)
+        self.lower_menu.opt_sel.set("Image")
+        for toggle in self.lower_menu.buttons:
+            if not toggle.get_status():
+                toggle.click()
+        self.lower_menu.distance_link.set(self.lower_menu.slider_values[0])
+        for i, link in enumerate(self.lower_menu.links):
+            link.set(self.lower_menu.slider_values[i + 1])
+
+        self.update_results = temp_func_update_r
+        self.diagram.update_l_lenses = temp_func_update_l
+        self.diagram.update_u_lenses = temp_func_update_u
+
+    def pass_function(self):
+        pass
```

Comment:
Rather than defining this here, maybe define it in reset_settings function. And maybe call it dummy_function.

Reply:
actually I kind of see why you did it this way, it isn't elegant but we can leave it I guess